front/lib/api/assistant/recent_authors.ts

```diff
@@ -0,0 +1,169 @@
+) {
+  const agentRecentAuthorIdsKey = _getRecentAuthorIdsKey({
+    agentId,
+    workspaceId,
+  });
+  await safeRedisClient(async (redis) => {
+    // Add <authorId:version> pairs to the sorted set, only if the version is greater than the one stored.
+    await redis.zAdd(agentRecentAuthorIdsKey, authorIdsWithScore, { GT: true });
+    // Set the expiry for the sorted set to manage its lifecycle.
+    await redis.expire(agentRecentAuthorIdsKey, recentAuthorIdsKeyTTL);
+  });
+}
+
+async function populateAuthorIdsFromDb(agentId: string, workspaceId: string) {
+  const recentAuthorIdsWithVersion = await fetchRecentAuthorIdsWithVersion(
+    agentId
+  );
+
+  if (recentAuthorIdsWithVersion.length === 0) {
+    return [];
+  }
+
+  const authorIdsWithScore = recentAuthorIdsWithVersion.map((a) => ({
+    // Redis only supports strings.
+    value: a.get("authorId").toString(),
+    score: a.version,
+  }));
+
+  await setAuthorIdsWithVersionInRedis(
+    agentId,
+    workspaceId,
+    authorIdsWithScore
+  );
+
+  return recentAuthorIdsWithVersion.map((a) => a.authorId.toString());
+}
+
+function renderAuthors(
+  authorIds: readonly string[],
+  members: UserType[],
+  currentUserId?: number
+): readonly string[] {
+  return (
+    authorIds
+      .map((id) => parseInt(id, 10))
+      .map((authorId) => {
+        // If authorId is the current requester, return "Me".
+        if (authorId === currentUserId) {
+          return "Me";
+        }
+        return members.find((m) => m.id === authorId)?.fullName ?? null;
+      })
+      // Filter out `null` authors.
+      .filter((name): name is string => name !== null)
+  );
+}
+
+export async function getAgentRecentAuthors(
+  {
+    agentConfiguration,
+    auth,
+  }: {
+    agentConfiguration: AgentConfigurationType;
+    auth: Authenticator;
+  },
+  members: UserType[]
+): Promise<AgentRecentAuthors> {
+  const { sId: agentId, versionAuthorId } = agentConfiguration;
+
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Owner is required");
+  }
+  const { sId: workspaceId } = owner;
+  const currentUserId = auth.user()?.id;
+
+  const isGlobalAgent = versionAuthorId === null;
+  // For global agents, which have no authors, return early.
+  if (isGlobalAgent) {
+    return [];
+  }
+
+  const agentRecentAuthorIdsKey = _getRecentAuthorIdsKey({
+    agentId,
+    workspaceId,
+  });
+
+  let recentAuthorIds = await safeRedisClient(async (redis) =>
+    redis.zRange(agentRecentAuthorIdsKey, 0, 2, { REV: true })
+  );
+
+  if (recentAuthorIds.length === 0) {
+    // Populate from the database and store in Redis if the entry is not already present.
+    recentAuthorIds = await populateAuthorIdsFromDb(agentId, workspaceId);
+  }
+
+  // Consider moving this logic to the FE if we need to fetch members in different places.
+  return renderAuthors(recentAuthorIds, members, currentUserId);
+}
+
+export async function agentConfigurationWasUpdatedBy({
```

Comment:
let's take `auth: Authenticator` and `agent: AgentCongifurationType` here to make things more "typed"